IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -308,10 +321,22 @@ - (MSIDIdToken *)getIDTokenForAccount:(MSIDAccountIdentifier *)accountIdentifier
     query.clientId = configuration.clientId;
     query.credentialType = MSIDIDTokenType;
 
-    return (MSIDIdToken *) [self getTokenWithAuthority:configuration.authority
-                                            cacheQuery:query
-                                               context:context
-                                                 error:error];
+    __auto_type idToken = (MSIDIdToken *)[self getTokenWithAuthority:configuration.authority
+                                                          cacheQuery:query
+                                                             context:context
+                                                               error:error];
+    
+    if (idToken)
+    {
+        MSID_LOG_INFO(context, @"Found id token for account %@.", _PII_NULLIFY(accountIdentifier));
+        MSID_LOG_INFO_PII(context, @"Found id token %@ for account %@.", idToken.rawIdToken, accountIdentifier);
```

Comment:
do we want to log entire id token here?

Reply:
I think it might be useful, aren't we allowed to log it if PII enabled?

Reply:
As we discussed offline, id token might be used as a credential. Decided to print only hash of it (same as access token).